civis/cli/_cli_commands.py

```diff
@@ -156,6 +165,55 @@ def _str_table_result(cols, rows):
     return '\n'.join(tb_strs)
 
 
+@click.command(
+    'follow-log',
+    help='Output live log from the most recent job run.' + _FOLLOW_LOG_NOTE)
+@click.argument('id', type=int)
+def jobs_follow_log(id):
+    client = civis.APIClient()
+    runs = client.jobs.list_runs(id, limit=1, order='id', order_dir='desc')
+    if not runs:
+        raise click.ClickException('No runs found for that job ID.')
+    run_id = runs[0].id
+    print('Run ID: ' + str(run_id))
+    _jobs_follow_run_log(id, run_id)
+
+
+@click.command(
+    'follow-run-log',
+    help='Output live run log.' + _FOLLOW_LOG_NOTE)
+@click.argument('id', type=int)
+@click.argument('run_id', type=int)
+def jobs_follow_run_log(id, run_id):
+    _jobs_follow_run_log(id, run_id)
+
+
+def _jobs_follow_run_log(id, run_id):
+    client = civis.APIClient(return_type='raw')
+    seen_max_log_id = 0
+    continue_polling = True
+
+    while continue_polling:
+        response = client.jobs.list_runs_logs(id, run_id,
+                                              last_id=seen_max_log_id)
+        if 'civis-max-id' in response.headers:
+            remote_max_log_id = int(response.headers['civis-max-id'])
+        log_finished = response.headers['civis-cache-control'] != 'no-store'
```

Comment:
A lot of these items being pulled out of headers are mysterious to me. It would be helpful to have a comment saying what is being pulled out of the header and what the value you're expecting means.

Reply:
I guess the meaning of the value here doesn't really matter as much as the fact that the 'no-store' means Platform is no longer expecting to get any more logs for this run.
Is there a variable name that would make it clearer?

Reply:
The variable name makes sense to me, but I find the names of things in the Platform response confusing (`civis-cache-control`, `no-store`). I can take your word for it that this variable holds a bool of whether the log is done or not, but I wondered if there was any context you could add about what Platform is passing. This could provide extra context if, for example, I needed to modify the code in the future for some reason.

Reply:
Currently it's either `max-age=120` or `no-store`, where the latter means that Platform has decided it might still be getting more logs. The `max-age=120` value is currently not used directly by the web UI either. It's in the same format as a [regular Cache-Control header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) to tell the client how long it can cache the results, but right now it's just used to tell the client when to stop polling for results.